include/rocksdb/sst_file_manager.h

```diff
@@ -64,22 +65,23 @@ class SstFileManager {
 // @param info_log: If not nullptr, info_log will be used to log errors.
 //
 // == Deletion rate limiting specific arguments ==
-// @param trash_dir: Path to the directory where deleted files will be moved
-//    to be deleted in a background thread while applying rate limiting. If this
-//    directory doesn't exist, it will be created. This directory should not be
-//    used by any other process or any other SstFileManager, Set to "" to
-//    disable deletion rate limiting.
+// @param trash_dir: Deprecated, this argument have no effect
 // @param rate_bytes_per_sec: How many bytes should be deleted per second, If
 //    this value is set to 1024 (1 Kb / sec) and we deleted a file of size 4 Kb
 //    in 1 second, we will wait for another 3 seconds before we delete other
 //    files, Set to 0 to disable deletion rate limiting.
 // @param delete_existing_trash: If set to true, the newly created
-//    SstFileManager will delete files that already exist in trash_dir.
+//    SstFileManager will delete files that already marked as trash in db_path.
 // @param status: If not nullptr, status will contain any errors that happened
 //    during creating the missing trash_dir or deleting existing files in trash.
+// @param paths: If passed, we will check if there are any files that are marked
+//    as trash in the passed paths so that they will schedule there deletes.
+//    This argument is used to clean up if the DB was shutdown without deleting
+//    all trash files.
 extern SstFileManager* NewSstFileManager(
     Env* env, std::shared_ptr<Logger> info_log = nullptr,
     std::string trash_dir = "", int64_t rate_bytes_per_sec = 0,
-    bool delete_existing_trash = true, Status* status = nullptr);
+    bool delete_existing_trash = true, Status* status = nullptr,
+    std::vector<std::string> paths = {});
```

Comment:
Do we consider still keep the `trash_dir`. By asking users to list all the paths (means DBs), it increases the overhead of manageability. If users have written the code in a way that they first create SST file managers and then open DB one by one by passing the object to the options, they'll need a dramatic code change to make it work.

Also, silently deprecating `trash_dir` is not a good idea. If people just upgrade to a new RocksDB release without reading the release notes, we may leave some files behind in that directory forever, which is not good. It doesn't even consider the case where users have an outside service cleaning up the trash dir.

Reply:
The problem with trash directory is that we need a different trash directory for every device, so instead we just keep files in the same place and rename them to ".trash"

I agree that we have 2 problems
- On upgrade, users may end up with a trash directory that wont be cleaned up
- Users need to pass the paths which increase manageability (this is only used for clean up on startup)

On possible solution would to be keep the trash directory but instead of moving files there, we just create symlinks to the trash files, and on restart we read all the symlinks and delete the actual files. does that sound like a good solution ?